handlers/mtn/mtn.go

```diff
@@ -120,46 +120,44 @@ func (h *handler) Send(ctx context.Context, msg courier.Msg, clog *courier.Chann
 		return nil, err
 	}
 
-	partSendURL, _ := url.Parse(sendURL)
+	baseURL := msg.Channel().StringConfigForKey(configAPIHost, apiHostURL)
+	partSendURL, _ := url.Parse(fmt.Sprintf("%s/%s", baseURL, "v2/messages/sms/outbound"))
 
 	status := h.Backend().NewMsgStatusForID(msg.Channel(), msg.ID(), courier.MsgErrored, clog)
-	parts := handlers.SplitMsgByChannel(msg.Channel(), handlers.GetTextAndAttachments(msg), maxMsgLength)
-	for i, part := range parts {
-		mtMsg := &mtPayload{}
-		mtMsg.From = strings.TrimPrefix(msg.Channel().Address(), "+")
-		mtMsg.To = []string{strings.TrimPrefix(msg.URN().Path(), "+")}
-		mtMsg.Message = part
-
-		requestBody := &bytes.Buffer{}
-		json.NewEncoder(requestBody).Encode(mtMsg)
-
-		// build our request
-		req, err := http.NewRequest(http.MethodPost, partSendURL.String(), requestBody)
-		if err != nil {
-			return nil, err
-		}
-		req.Header.Set("Content-Type", "application/json")
-		req.Header.Set("Accept", "application/json")
-		req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", accessToken))
-
-		resp, respBody, err := handlers.RequestHTTP(req, clog)
-		if err != nil || resp.StatusCode/100 != 2 {
-			return status, nil
-		}
-
-		externalID, err := jsonparser.GetString(respBody, "transactionId")
-		if err != nil {
-			clog.Error(courier.ErrorResponseValueMissing("transactionId"))
-			return status, nil
-		}
-
-		// if this is our first message, record the external id
-		if i == 0 {
-			status.SetExternalID(externalID)
-			status.SetStatus(courier.MsgWired)
-		}
 
+	mtMsg := &mtPayload{}
```

Comment:
we do not need to divide the message in parts, they do

Reply:
and since we need the clientCorrelator to be unique for each message, I used the UUID

I am not sure how our retry mechanism  in case of error will work but we'll see

Reply:
So `clientCorrelator` is like an external id except we can generate it? `Msg.UUID` isn't indexed so we don't want to lookup msgs by that. But we could put msg.id in `clientCorrelator` and just use that for lookups.